tests/test_bytes.rs

```diff
@@ -514,3 +514,103 @@ fn partial_eq_bytesmut() {
     assert!(bytes2 != bytesmut);
     assert!(bytesmut != bytes2);
 }
+
+#[test]
+fn unsplit_basic() {
+    let mut buf = BytesMut::with_capacity(64);
+    buf.extend_from_slice(b"aaabbbcccddd");
+    
+    let splitted = buf.split_off(6);
+    assert_eq!(b"aaabbb", &buf[..]);
+    assert_eq!(b"cccddd", &splitted[..]);
+    
+    buf.unsplit(splitted);
+    assert_eq!(b"aaabbbcccddd", &buf[..]);
+}
+
+#[test]
+fn unsplit_inline_arc() {
+    let mut buf = BytesMut::with_capacity(8); //inline
+    buf.extend_from_slice(b"aaaabbbb");
+    
+    let mut buf2 = BytesMut::with_capacity(64); 
+    buf2.extend_from_slice(b"ccccddddeeee");
+    
+    let _buf3 = buf2.split_off(8); //arc
```

Comment:
What's the purpose of `_buf3`?

Reply:
A good point, it does not really make the test any clearer for the reader. Now removed all of them.